cli/azd/pkg/input/console.go

```diff
@@ -63,14 +67,35 @@ func (c *AskerConsole) SetWriter(writer io.Writer) {
 
 // Prints out a message to the underlying console write
 func (c *AskerConsole) Message(ctx context.Context, message string) {
-	// Only write to the console during interactive & non-formatted responses.
-	if c.interactive && (c.formatter == nil || c.formatter.Kind() == output.NoneFormat) {
+	if c.formatter != nil && c.formatter.Kind() == output.JsonFormat {
+		// we do not expect formatter.Format to fail on valid input. Discard the error to appease the linter (note
+		// that we implicitly discard the error from fmt.Fprintln later, as well).
+		_ = c.formatter.Format(c.consoleMessageForMessage(message), c.writer, nil)
```

Comment:
Did you try writing a simple app to parse the output of ​infra create --output json​?

What should be the algorithm be to detect the "resulting machine readable output" of the command?